### PR TITLE
docs: add JSDoc for some exports @W-13278716

### PR DESCRIPTION
**Pull request by wjhsf**
## Details

This time, with actual things to read!

## Does this pull request introduce a breaking change?

* ✅ No, it does not introduce a breaking change.
* 🚨 Yes, it does introduce a breaking change.

## Does this pull request introduce an observable change?

* ✅ No, it does not introduce an observable change.
* ⚠️ Yes, it does include an observable change.

### CODE REVIEW

**Review by nolanlawson — APPROVED** [review]
Fantastic work.

File: packages/@lwc/errors/src/compiler/utils.ts [hunk 1]
```diff
@@ -58,6 +58,11 @@ export class CompilerError extends Error implements CompilerDiagnostic {
     }
 }
 
+/**
+ * Extracts an error code from the given error.
+ * @param error The error to check.
+ * @returns The found error code.
```
Comment (ekashida) [inline packages/@lwc/errors/src/compiler/utils.ts:64]:
```suggestion
 * @returns The error code, if found.
```

File: packages/@lwc/errors/src/compiler/errors.ts [hunk 2]
```diff
@@ -85,6 +93,13 @@ export function generateCompilerError(
     return error;
 }
 
+/**
+ * Validates that the provided condition is truthy.
+ * @param condition Condition to check.
+ * @param errorInfo The object holding the error metadata.
+ * @param args Values used to fill the error message template.
+ * @throws Throws a compiler error if the provided condition is falsy.
```
Comment (ekashida) [inline packages/@lwc/errors/src/compiler/errors.ts:101]:
```suggestion
 * @throws Throws a compiler error if the provided condition is false.
```

File: packages/@lwc/errors/src/compiler/errors.ts [hunk 3]
```diff
@@ -85,6 +93,13 @@ export function generateCompilerError(
     return error;
 }
 
+/**
+ * Validates that the provided condition is truthy.
```
[the remark this answers is not in the document]
Comment (wjhsf) [reply packages/@lwc/errors/src/compiler/errors.ts:97]: The function checks for truthiness, not for `=== true`, so I think "truthy"/"falsy" are more accurate here.

[the remark this answers is not in the document]
Comment (ekashida) [reply packages/@lwc/errors/src/compiler/errors.ts:97]: That's fine, I just saw that the type was `boolean` so we would only get either `true` or `false`

**Pull request merged**